scripts/SREP/infra-node-resize/script.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+# This script checks if any tech preview features are enabled through feature sets.
+
+set -e
+set -o nounset
+set -o pipefail
+
+# fetch the infra machinepools
+get_machinepool() {
+    echo 'FETCHING INFRA MACHINEPOOLS...'
+    NAMESPACE=$(oc get ns | grep "$CLUSTER_ID" | awk '{print $1}')
+    oc -n "$NAMESPACE" get machinepool
+    INFRA_MACHINEPOOL=$(oc -n "$NAMESPACE"  get machinepool -o json | jq -r '.items[] | select(.spec.name=="infra") | .metadata.name')
+    oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r ".spec.platform.type"
+}
+
+# update the yaml file
+update_cluster_infra_yaml() {
+    echo 'UPDATING THE YAML FILE FOR TEMPORARY MACHINEPOOL...'
+    oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o yaml > /tmp/cluster-infra.yaml
+    yq -i 'del(.metadata.creationTimestamp) | del(.metadata.finalizer) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.metadata.uid) | del(.status)' /tmp/cluster-infra.yaml
+    cat /tmp/cluster-infra.yaml 
+    CLOUD=$(oc -n "$NAMESPACE" get machinepool "$INFRA_MACHINEPOOL" -o json | jq -r '.spec.platform | keys_unsorted[0]')
+    yq -i ".metadata.name |= . + 2 | .spec.name |= . + 2 | .spec.platform.$CLOUD.type |= env(INSTANCE_SIZE)" /tmp/cluster-infra.yaml 
+}
+
+# create the temporary machinepool
+create_temp_machinepool() {
+    echo 'CREATING TEMPORARY MACHINEPOOL...'
+    oc -n "$NAMESPACE" create -f /tmp/cluster-infra.yaml
+    while [[ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ]];
+    do
+        echo 'MACHINES ARE STILL NOT CORRECTLY PROVISIONED...'
+        timeout 10
+    done
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# delete the original machinepool
+delete_original_machinepool() {
+    echo 'DELETING THE ORIGINAL MACHINEPOOL...'
+    oc scale --replicas 0 machinepool "$INFRA_MACHINEPOOL" -n "$NAMESPACE"
+    oc -n "$NAMESPACE" get machinepool
+    while [[ ! -z "$(oc get machinepool -n "$NAMESPACE" "$INFRA_MACHINEPOOL" -o=jsonpath='{.status.machineSets[0].errorMessage}')" ]];
+    do
+        echo 'MACHINES ARE STILL NOT PROPERL DELETED...'
+        timeout 10
+    done
+    oc -n "$NAMESPACE" delete machinepool "$INFRA_MACHINEPOOL"
+    echo 'ORIGINAL MACHINEPOOLS WHICH ARE NOT RESIZED ARE DELETED...'
+    oc -n "$NAMESPACE" get machinepool
+}
+
+# update yaml file cluster 
+update_new_infra_yaml() {
+    echo 'UPDATING YAML FILE...'
+    yq eval '.metadata.name |= sub("2$"; "") | .spec.name |= sub("2$"; "")' -i /tmp/cluster-infra.yaml 
```

Comment:
will it be better if we build the yaml for the new machinepool separately?
so we do not need to update the same file twice.

Reply:
means we can build two yaml from the func `update_cluster_infra_yaml` maybe change the name to `build_cluster_infra_yaml`
one for the name with prefix -2 and one for the original name, both have the same size.
And in the following steps, we just need to create `-2` and delete original and create new-original and delete `-2`.

Reply:
Good idea! But I think we can keep it as it is, both are in separate functions and changing the yaml files separately. It won't change the code complexity.